### PR TITLE
new app submission for domestique-baston

### PR DESCRIPTION
**Pull request by glevner**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.itch.domestique-baston.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+command: domestique-baston
+modules:
+  - name: domestique-baston
+    buildsystem: simple
+    build-commands:
+      - install -D 'domestique-baston-gles3' /app/bin/domestique-baston
+    sources:
+      - type: file
+        path: 'domestique-baston-gles3'
+finish-args:
```
Comment (hfiguiere) [inline io.itch.domestique-baston.yml:14]: this is usually above `modules`

File: io.itch.domestique-baston.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline io.itch.domestique-baston.yml:17]: does it really support wayland? If yes then it's ` fallback-x11` above.

File: io.itch.domestique-baston.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+sdk: org.freedesktop.Sdk
+command: domestique-baston
+modules:
+  - name: domestique-baston
+    buildsystem: simple
+    build-commands:
+      - install -D 'domestique-baston-gles3' /app/bin/domestique-baston
+    sources:
+      - type: file
+        path: 'domestique-baston-gles3'
```
Comment (hfiguiere) [inline io.itch.domestique-baston.yml:13]: this file doesn't exist. and is should be a `url` because upstream build artifacts don't belong here anyway

**Review by hfiguiere — CHANGES_REQUESTED** [review]
there is no .desktop, icon and appstream file.

this can not be built.

File: io.itch.domestique-baston.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+app-id: io.itch.domestique-baston
```
Comment (hfiguiere) [inline io.itch.domestique-baston.yml:1]: this app-id is not acceptable as you don't own `io.itch`. Like for github, you need the user/org. There are other `io.itch` based packages that do that.

**Comment by glevner** [comment]
Thanks for all the comments. The manifest is the only file in the branch,
for now, because the instructions
<https://github.com/flathub/flathub/wiki/App-Submission#how-to-submit-an-app>
only say to commit the manifest... I will add the others.

As for the source file, it is generated for Linux by the Godot
<https://godotengine.org/> engine from a Godot project. I am not sure how
else to go about it. Any suggestions would be welcome...

Thanks,
Geoff

File: io.itch.domestique-baston.yml (hunk 4 above)
Comment (glevner) [reply io.itch.domestique-baston.yml:1]: So the app-id should be `io.itch.domestique-baston.domestique-baston`?

**Comment by glevner** [comment]
I see that I should probably be using the Godot BaseApp and supplying a .pck file rather than the executable. But I cannot get the BaseApp to work. Again, any tips would be most welcome.

File: domestique-baston-gles3
Comment (hfiguiere) [inline domestique-baston-gles3:1]: this file do not belong here. 

build artifacts or source code should come from the upstream project.

File: io.itch.domestique_baston.domestique_baston.desktop [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Name=Domestique Baston
+Comment=A 1 vs. 1 fighting game
+Type=Application
+Exec=io.itch.domestique_baston.domestique_baston
```
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.desktop:5]: this not the name of the executable as installed.

File: io.itch.domestique_baston.domestique_baston.png
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.png:1]: this file is never installed. Also the image size is not standard for icons

File: io.itch.domestique_baston.domestique_baston.metainfo.xml
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.metainfo.xml:1]: this file is never installed.

File: io.itch.domestique_baston.domestique_baston.desktop
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.desktop:1]: this file is never installed.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
icon, .desktop and appstream files should be submitted upstream.

**Comment by hfiguiere** [comment]
also you still try to build an aarch64 flatpak with a binary that is likely to not be.

File: io.itch.domestique_baston.domestique_baston.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  - --socket=pulseaudio
+modules:
+  - name: domestique_baston
+    buildsystem: simple
+    build-commands:
+      - install -D DOMESTIQUE_BASTON_gles3.x86_64 /app/bin/domestique_baston
+      - install -D -m 0644 *.pdf /app/bin
+      - install -D -m 0644 io.itch.domestique_baston.domestique_baston.metainfo.xml /app/share/metainfo/io.itch.domestique_baston.domestique_baston.metainfo.xml
+      - install -D -m 0644 io.itch.domestique_baston.domestique_baston.desktop /app/share/applications/io.itch.domestique_baston.domestique_baston.desktop
+      - install -D -m 0644 icon96.png /app/share/icons/hicolor/96x96/apps/io.itch.domestique_baston.domestique_baston.png
```
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.yml:19]: minimum required size is 128x128. Or a SVG.

File: io.itch.domestique_baston.domestique_baston.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      - install -D -m 0644 icon48.png /app/share/icons/hicolor/48x48/apps/io.itch.domestique_baston.domestique_baston.png
+    sources:
+      - type: archive
+        url: 'http://www.ferdiferdi.com/db/DOMESTIQUE_BASTON_LINUX.zip'
```
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.yml:23]: the URL is not versioned which mean that when there is a new release, things will break. It's a bad practice.

File: io.itch.domestique_baston.domestique_baston.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+      It pictures a couple fighting each other at home. That’s the reason why
+      there is only 1 background and 2 playable characters.
+    </p>
+    <p>
+      This game is in French slang just because it’s funny. No other reason.
+      Besides, who really cares about, for instance, the story or even what’s
+      told in such a game?
+    </p>
+  </description>
+</component>
```
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.metainfo.xml:31]: at least one screenshot is required.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
I know it will fail

bot, build io.itch.domestique_baston.domestique_baston

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Appstream file, icons and .desktop file should be part of the distribution.

File: io.itch.domestique_baston.domestique_baston.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  - --share=ipc
+  - --socket=x11
+  - --device=all
+  - --socket=pulseaudio
+modules:
+  - name: domestique_baston
+    buildsystem: simple
+    build-commands:
+      - install -D DOMESTIQUE_BASTON_gles3.x86_64 /app/bin/domestique_baston
+      - install -D -m 0644 *.pdf /app/bin
```
Comment (hfiguiere) [inline io.itch.domestique_baston.domestique_baston.yml:16]: not sure what you expect the user to be able to do with these PDF files.

**Comment by glevner** [comment]
> Appstream file, icons and .desktop file should be part of the distribution.

I don't know what that means?

File: io.itch.domestique_baston.domestique_baston.yml (hunk 9 above)
Comment (glevner) [reply io.itch.domestique_baston.domestique_baston.yml:16]: It's documentation. I can remove it, or put it somewhere else, but where?

Comment (hfiguiere) [reply io.itch.domestique_baston.domestique_baston.yml]: they are not accessible, so unless you have a PDF viewer they are doing nothing.

**Comment by hfiguiere** [comment]
bot, build io.itch.domestique_baston.domestique_baston

**Pull request closed without merging**